index.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -e
+
+# Override the github.* context env vars with the ones that we got from the user
+# that may (or may not be) different from the original $GITHUB_* env vars.
+export GITHUB_TOKEN="$INPUT_TOKEN"
+export GITHUB_SERVER_URL="$INPUT_GITHUB_SERVER_URL"
+export GITHUB_REPOSITORY="$INPUT_REPOSITORY"
+
+# This is the default host that gh uses for clones and commands without a repo
+# context (a .git folder). We use Bash string magic to get the github.com part
+# from a full origin (no pathname) like https://github.com -> github.com
+export GH_HOST="${GITHUB_SERVER_URL#*//}"
+
+echo '🟪 Configuring local Git using gh auth setup-git...'
+gh auth setup-git
+
+# https://stackoverflow.com/a/28180781
+echo "🟪 Cloning $GITHUB_REPOSITORY.wiki into $INPUT_PATH..."
+cd "$INPUT_PATH"
+gh repo clone "$GITHUB_REPOSITORY.wiki" .git -- --bare
+trap 'rm -rf .git' SIGINT SIGTERM ERR EXIT
```

Comment:
is this a good cleanup operation, or is there a better way? bash doesn't really have try-finally

Reply:
Perhaps we can just naively cleanup at the end without any "try-finally" behavior. The majority of workflows would stop if this action fails, which basically "cleans up," except possibly in a self-hosted runner. On failure, the `.git` folder can continue to exist for the user's debug purposes.